src/binary_tree_partition.jl

```diff
@@ -143,9 +152,8 @@ function _remove_deltas(tn::ITensorNetwork, deltas::Vector{ITensor})
       _root_union!(ds, find_root!(ds, i2), find_root!(ds, i1))
     end
   end
-  tn = map_data(
-    t -> replaceinds(t, deltainds => [find_root!(ds, i) for i in deltainds]), tn; edges=[]
-  )
+  sim_deltainds = [find_root!(ds, i) for i in deltainds]
```

Comment:
I got confused by the the function name `_remove_deltas`, maybe we could call it `_contract_deltas`?

Also, it makes more sense to me to just pass a single tensor network, i.e.:
```julia
function _contract_deltas(tn::ITensorNetwork)
  # [...]
end
```
and then analyze which tensors are deltas. It looks like we don't have a function for that yet in NDTensors/ITensors but we could define:
```julia
is_delta(it::ITensor) = is_delta(tensor(it))
is_delta(t::Tensor) = false
is_delta(t::UniformDiagTensor) = isone(getdiagindex(t, 1))
```
(for now in ITensorNetwork.jl and then we can move it to NDTensors.jl/ITensors.jl). Note that `UniformDiagTensor` is defined in `NDTensors` and may not be exported. Also, it would be fairly easy to extend to the case where the diagonal elements aren't one (i.e. a scaled delta tensor) by scaling the network by the uniform diagonal values but we could leave that for future work since that isn't a common situation anyway.

Reply:
Yes renaming it to `_contract_deltas` makes sense, `_remove_deltas` is a bit confusing.

I'm actually also thinking about the name `_simplify_deltas`, so the function would be 
```julia
function _simplify_deltas(tn::ITensorNetwork)
  # [...]
  return new_tn, simplified_deltas
end
```
For example, when the input `tn` contains tensors `randomITensor(i), randomITensor(j), delta(k,l), delta(l,m)`, the output `new_tn` will contain `randomITensor(i), randomITensor(j)` and `splited_deltas` contains `delta(k,m)`.

The reason we want to split `simplified_deltas` and `new_tn` in the function output is that there can be new tensors in 
`simplified_deltas`, and to add them into `new_tn` we need to give each of them a vertex name. It's not easy to have a general naming convention to me so I think it's easier to split it out from `new_tn`.

Reply:
actually please ignore my last message, there are simple ways to fix my concern. The AP
```julia
function _contract_deltas(tn::ITensorNetwork)
  # [...]
  return new_tn
end
```
should work well.

Reply:
@mtfishman Refactoring `_contract_deltas` involves multiple changes, so I opened a new PR for that: #73 
Once that is merged I will then update this PR.